lib/win32/certstore.rb

```diff
@@ -31,18 +31,19 @@ class Certstore
 
     attr_accessor :store_name
 
-    def initialize(store_name)
+    def initialize(store_name, store_location)
       @store_name = store_name
-      @certstore_handler = open(store_name)
+      @store_location = store_location
+      @certstore_handler = open(store_name, store_location)
     end
 
     # To open given certificate store
-    def self.open(store_name)
+    def self.open(store_name, store_location)
       validate_store(store_name)
       if block_given?
-        yield new(store_name)
+        yield new(store_name, store_location)
       else
-        new(store_name)
+        new(store_name, store_location)
```

Comment:
call this with `new(store_name, store_location: store_location)` in both cases (and anywhere else I've missed it)